carve-lockpicks.lic

```diff
@@ -0,0 +1,114 @@
+=begin
+	Need these settings in your yaml:
+		#Stow locations for Grandmaster's, Master's, and anything less.
+    grand_container: backpack
+    master_container: haversack
+    trash_container: cloak
+    #Whether to initial Grandmaster's with an extra carve.
+    initial: true
+=end
+
+custom_require.call(%w[common common-items common-crafting])
+
+class CarveLockpicks
+
+  def initialize
+    @settings = get_settings
+    @bag = @settings.crafting_container
+    @belt = @settings.engineering_belt
+	  @grandbag = @settings.grand_container
+	  @masterbag = @settings.master_container
+	  @trashbag = @settings.trash_container
+	  @initial = @settings.initial
+    @tool = "carving knife"
+    DRC.wait_for_script_to_complete('buff', ['carve'])
+    DRC.bput('sit', 'You sit', 'You are already sitting', 'You rise', 'While swimming?') unless sitting?
+    DRCC.get_crafting_item("carving knife", @crafting_container, @crafting_container_items, @belt)
+    if DRCI.exists?("keyblank pocket")
+      get_keyblank
+    else
+      DRCC.stow_crafting_item(@tool, @bag, @belt)
+      exit
+	end
+  end
+
+
+  def get_keyblank
+    case DRC.bput "get keyblank from my keyblank pocket", 'You get', 'What were you referring to?'
+    when /What were you referring to?/
+      case DRC.bput "count my keyblank pocket", /There's nothing inside the keyblank pocket/, /It looks like there/, /I could not find what you were referring to./
+      when /There's nothing inside the keyblank pocket/
+        empty_pocket
+      when /It looks like there/
+        DRC.bput "open my keyblank pocket", 'You open a'
+      when /I could not find what you were referring to./
+        DRCC.stow_crafting_item(@tool, @belt, @bag)
+        exit
+      end
+	    get_keyblank
+    when /You get/
+      carve_keyblank
+    end
+  end
+
+  def carve_keyblank
+    case DRC.bput('carve my keyblank with my knife', /proudly glance down at a grandmaster/, /begin to carefully carve/, /proudly glance down at a master/, /but feel your knife slip/)
+    when /proudly glance down at a grandmaster/
+      initial_grand if @initial
+	    stow_item("grand")
+    when /begin to carefully carve/
+      carve_lockpick
+    when /proudly glance down at a master/
+      stow_item("master")
+    when /but feel your knife slip and the iron keyblank snaps like a twig/
+      get_keyblank
+    end
+  end
+
+  def carve_lockpick
+    case DRC.bput('carve my lockpick with my knife', /proudly glance down at a grandmaster/, /proudly glance down at a master/, /into the side of a/, /find a creature to carve/, /It would be better/, /Working with extreme caution/, /but feel your knife slip/)
+    when /proudly glance down at a grandmaster/
+      initial_grand if @initial
```

Comment:
What do you think about moving the conditional into a guard clause in `initial_grand`?

Reply:
I think I understand what you mean. Updated, it's now under stow_item, where it carves again IF @initial_grand = true

Reply:
Which removes that method, forgot that part. Second updated version is correct.

Reply:
What I meant was a guard clause in the `initial_grand` function:

```
def initial_grand
  return unless @initial
  ...
end
```

But I'm fine with the current approach, as long as you remove line 71 (which calls the now removed function).

Reply:
It's being stubborn, but I think it's gone now.